hooks/image/51-ek-content-list

```diff
@@ -0,0 +1,24 @@
+# Populate the Endless Key home directory
+
+if [ -z "${EIB_ENDLESSKEY_COLLECTIONS}" ]; then
+  exit 0
+fi
+
+if [[ ! "${EIB_FLATPAK_REMOTE_FLATHUB_APPS}" =~ .*"org.endlessos.Key".* ]]; then
+  exit 0
+fi
+
+# This file is used by hooks/image/52-ek-content-cache and
+# hooks/image/53-ek-content-preload
+EK_CHANNELS_FILE="${EIB_TMPDIR}"/ek-channels
+
+rm -f "${EK_CHANNELS_FILE}"
+collections="${OSTREE_VAR}"/lib/flatpak/app/org.endlessos.Key/current/active/files/share/endless-key/collections/*.json
+for collection in ${collections}; do
+  # Check if the file basename stripped off of -0001.json is part of the list
+  # of collections to be installed.
+  bn=$(basename ${collection})
+  if [[ "${EIB_ENDLESSKEY_COLLECTIONS}" =~ .*"${bn%-????.json}".* ]] ; then
+    jq -r '.channels[].id' "${collection}" >> "${EK_CHANNELS_FILE}"
+  fi
+done
```

Comment:
It would be nice to do the sorting once here. Like:

```
rm -f "${EK_CHANNELS_FILE}" "${EK_CHANNELS_FILE}.tmp"
for collection in ${collections}; do
  some channel filtering >> "${EK_CHANNELS_FILE}.tmp"
done
if [ -f "${EK_CHANNELS_FILE}.tmp" ]; then
  sort -u "${EK_CHANNELS_FILE}.tmp" > "${EK_CHANNELS_FILE}"
fi
```

Reply:
Yes, good point! So we don't try to cache / import a channel more than once. I'll change that.